Automattic/wp-calypso
# GuidedTours: Use JSX for main tour

Split from #7030

Aims to be a more minimal PR, so I've stripped out:
 - analytics
 - blanket exit
 - positioning fixes
 - themes tour

There should be more to strip out, that we can put back in — in later PRs. I think we should concentrate on functional feature parity with the old approach first. And also reduce review overhead.

TODO:
- [x] Completely remove dependency on old guided tours config
- [x] Decide where to put `wait` code — *just delete it for now*
- [x] Update PropTypes for config elements
- [x] Tidy
- [x] Squash

To test:
- go to http://calypso.localhost:3000 and click around — just check whether anything is obviously broken
- go to http://calypso.localhost:3000/?tour=main — check whether you can proceed through the main tour (see below) properly
- also check whether exiting the tour (via the No Thanks / Quit buttons) works

GIF of the main tour as it should appear: 

![main-tour](https://cloud.githubusercontent.com/assets/23619/17583111/a134f678-5faf-11e6-81b5-8c153232d50f.gif)

Test live: https://calypso.live/?branch=update/guidedtours-framework-to-jsx
